src/ddmd/mars.d

```diff
@@ -2156,7 +2156,10 @@ private bool parseCommandLine(const ref Strings arguments, const size_t argc, re
                         error("-run must be followed by a source file, not '%s'", arguments[i + 1]);
                         break;
                     }
-                    files.push(arguments[i + 1]);
+                    if (strcmp(arguments[i + 1], "-") == 0)
```

Comment:
This is too simplistic, it won't work if there's a command after `-run`, e.g. `dmd -run -m64 -`

Reply:
Actually, `-run` takes an argument, so your example is invalid.  (I'm not sure why whoever originally implemented `-run` did it this way, but that's how it currently works, and this PR is just following in its steps.)

Reply:
Oh sorry. I just looked over this again from my phone and imagined that doing `dmd -run -version=Foo myfile.d` is legimate, so yeah that seems like bad design too me too, but that's an opportunity for another PR ;-)